sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AliasResolution.scala

```diff
@@ -52,6 +42,7 @@ object AliasResolution {
       case ne: NamedExpression => ne
       case go @ GeneratorOuter(g: Generator) if g.resolved => MultiAlias(go, Nil)
       case e if !e.resolved => u
+      case p: PipeExpression => resolve(u.copy(child = p.replacement))
```

Comment:
This makes me think that if we really want to make `PipeExpression` extend `RuntimeReplaceable`. `PipeExpression` is more like an unresolved expression that should be trimmed during analysis.

We can clean it up in a followup.

Reply:
Good point, I looked and was able to switch `PipeExpression` from `RuntimeReplaceable` to `Unevaluable` instead. Now `ResolveAliases` maintains full responsibility for checking the invariants and removing the `PipeExpression` once its child is resolved. I also synced from master and updated the golden files.